### ISSUE TITLE
Change modal headings to h3 and add aria-labelledby

### ISSUE THREAD

**Issue by alundgard** [sul-dlss/heracles-etd#243]
This change applies to all modals.

### Reported by SODA
> 
> [Summary]
> Dialog / Alert Dialog: Dialog is missing appropriate role and/or attributes
> 
> [Description]
> "The element appears and functions like a modal dialog but is missing required ARIA role(s) and/or attribute(s).
> Heading should be specified as the label for the modal. "
> 
> [Impact]
> Moderate
> 
> [Recommended to fix]
> "Change the heading to an h3 and programmatically associate it with the dialog. 
> 
> ```
> <div class=""modal modal-lg show"" tabindex=""-1"" id=""cc-licenses"" style=""display: block;"" aria-modal=""true"" role=""dialog"" aria-labelledby=""modalHeading"">
> <h3 id=""modalHeading"">Creative Commons Licenses</h3>
> ```
> 
> Rule : 
> The name, role, value, states, and properties of user interface components MUST be programmatically determinable by assistive technologies.
> 
> How To Fix: 
> A dialog is a window overlaid on either the primary window or another dialog window. Windows under a modal dialog are inert. That is, users cannot interact with content outside an active dialog window.
> 
> Fix this issue by using the following roles and attributes:
> 1. The element that serves as the dialog container has a role of dialog.
> 2. All elements required to operate the dialog are descendants of the element that has role dialog.
> 3. The dialog container element has aria-modal set to true.
> 4. The dialog has either:
> a. A value set for the aria-labelledby property that refers to a visible dialog title.
> b. A label specified by aria-label.
> 5. Optionally (if focus is set to an element other than the heading text inside the dialog) the aria-describedby property is set on the element with the dialog role to indicate which element or elements in the dialog contain content that describes the primary purpose or message of the dialog. Specifying descriptive elements enables screen readers to announce the description along with the dialog title and initially focused element when the dialog opens.
> 
> Reference: 
> WAI-ARIA Authoring Practices: https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/
> 
> Background: 
> Every user interface control must have a role along with any applicable states and properties so that screen reader users know how to interact with the control. Native HTML elements - such as <button>, <a>, <input>, <select> - already have a role, and their necessary states and properties - such as the checked/unchecked state of a checkbox - are automatically conveyed so nothing more needs to be done. If you create a custom version of a native HTML element or a custom control or widget that does not have a native HTML equivalent, you must add the relevant role(s) and any applicable states and properties using ARIA as well as expected keyboard interactions."
